### PR TITLE
Feature/316 tribal upstream watershed

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-316](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-316)

## Main Changes:
* Added the _Upstream Watershed_ widget to the **Tribe** page.
* Altered the widget behavior to allow the user to select a watershed to view its upstream.
* Included an instruction box for the widget.
* Changed text on the main **State & Tribal** page.

## Steps To Test:
1. Visit the Red Lake Band of Chippewa Indians page, i.e. [http://localhost:3000/tribe/REDLAKE](http://localhost:3000/tribe/REDLAKE).
2. Click the _Upstream Watershed_ widget.
3. Check that a close-able instruction box appears and that the widget icon indicates activity.
4. Zoom in until watershed boundary lines appear, and confirm that the instructions update.
5. Click inside a watershed and confirm that the upstream watershed is shown and positioned correctly.
6. Click the widget again, and confirm that the extent returns to its previous location and the upstream watershed disappears.
7. Activate the widget, then click outside the map; the widget should exit "selection mode". Alternatively, click the widget again to deactivate.
8. Activate the widget, then click an area in one of the great lakes to verify errors are handled properly.
9. Verify that the proper text has been changed on the **State & Tribal** page.

### CODE REVIEW

File: app/client/src/components/shared/MapWidgets.tsx [hunk 1]
```diff
@@ -1714,255 +1795,567 @@ function ShowSurroundingMonitoringLocations({
...
+        upstreamLayer.error = true;
+        upstreamLayer.graphics.removeAll();
+        setUpstreamLayer(upstreamLayer);
+        canDisable && setUpstreamWidgetDisabled(true);
+        setUpstreamLayerVisible(false);
+        setErrorMessage(
+          `Unable to get upstream watershed data for ${
+            huc12 ? 'the selected' : 'this'
+          } location.`,
+        );
```
Comment (cschwinderg) [inline app/client/src/components/shared/MapWidgets.tsx:1899]:
```suggestion
        setErrorMessage(
          `No upstream watershed data available for ${
            huc12 ? 'the selected' : 'this'
          } location.`,
        );
```

I think this should be a `No data available` message and be an info level message. This language sounds like there was a problem, when really there just isn't any data available. We should do the same thing for the community page as well.

I think it is expected for some locations to not have any upstream watersheds. Especially for watersheds in the desert. 


To reproduce go to http://localhost:3000/tribe/PUEBLO_SANTAANA and then follow the GIF below.
no-upstream-watershed

File: app/client/src/components/pages/StateTribal.js [hunk 2]
```diff
@@ -413,8 +413,8 @@ function StateTribal() {
             <label css={promptStyles} htmlFor="hmw-state-select-input">
               <strong>Let’s get started!</strong>&nbsp;&nbsp;
               <em>
-                Select your state or territory from the drop down to begin
-                exploring water quality.
+                Select your state, tribe, or territory from the drop down to
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/pages/StateTribal.js:416]: Sounds good! It did cross my mind.

**Comment by maxdiebold-erg** [comment]
@cschwinderg Thanks for the suggestions, those should be updated. I did dock the box at the bottom of the map for small screens, but I'm worried that users might miss that, depending on their scroll position. Initially, I wanted to keep it close to the widget icon. We should see what the clients think, but we may want to dock it at the top, instead.

**Comment by cschwinderg** [comment]
@maxdiebold-erg I'm good with docking it to the top.

**Review by cschwinderg — APPROVED** [review]
Looks good! Thanks for making those changes.

**Pull request merged**